plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -277,6 +298,105 @@ fieldSuggestCodeAction recorder ide _ (CodeActionParams _ _ (TextDocumentIdentif
+      uri = msgParam ^. JL.textDocument . JL.uri
+      isSectionArgName name (Syntax.Section _ sectionArgName _) = name == CabalFields.onelineSectionArgs sectionArgName
+      isSectionArgName _ _ = False
+      isModuleName name (_,  moduleName) = name == moduleName
+
+      lookupBuildTargetPackageDescription :: PackageDescription -> Maybe T.Text -> [BuildInfo]
+      lookupBuildTargetPackageDescription (PackageDescription {..}) Nothing =
+        case library of
+          Nothing                       -> error "Target is a main library but no main library was found"
+          Just (Library {libBuildInfo}) -> [libBuildInfo]
+      lookupBuildTargetPackageDescription (PackageDescription {..}) (Just buildTargetName) =
+        Maybe.catMaybes $
+          map (\exec -> executableNameLookup exec buildTargetName) executables <>
+          map (\lib -> subLibraryNameLookup lib buildTargetName) subLibraries <>
+          map (\lib -> foreignLibsNameLookup lib buildTargetName) foreignLibs <>
+          map (\test -> testSuiteNameLookup test buildTargetName) testSuites <>
+          map (\bench -> benchmarkNameLookup bench buildTargetName) benchmarks
+        where
+          executableNameLookup :: Executable -> T.Text -> Maybe BuildInfo
+          executableNameLookup (Executable {exeName, buildInfo}) buildTargetName =
+            if T.pack (unUnqualComponentName exeName) == buildTargetName
+              then Just buildInfo
+              else Nothing
+          subLibraryNameLookup :: Library -> T.Text -> Maybe BuildInfo
+          subLibraryNameLookup (Library {libName, libBuildInfo}) buildTargetName =
+            case libName of
+              (LSubLibName name) ->
+                if T.pack (unUnqualComponentName name) == buildTargetName
+                  then Just libBuildInfo
+                  else Nothing
+              LMainLibName -> Nothing
+          foreignLibsNameLookup :: ForeignLib -> T.Text -> Maybe BuildInfo
+          foreignLibsNameLookup (ForeignLib {foreignLibName, foreignLibBuildInfo}) buildTargetName =
+              if T.pack (unUnqualComponentName foreignLibName) == buildTargetName
+              then Just foreignLibBuildInfo
+              else Nothing
+          testSuiteNameLookup :: TestSuite -> T.Text -> Maybe BuildInfo
+          testSuiteNameLookup (TestSuite {testName, testBuildInfo}) buildTargetName =
+            if T.pack (unUnqualComponentName testName) == buildTargetName
+              then Just testBuildInfo
+              else Nothing
+          benchmarkNameLookup :: Benchmark -> T.Text -> Maybe BuildInfo
+          benchmarkNameLookup (Benchmark {benchmarkName, benchmarkBuildInfo}) buildTargetName =
+              if T.pack (unUnqualComponentName benchmarkName) == buildTargetName
+              then Just benchmarkBuildInfo
+              else Nothing
+
+      toHaskellFile :: T.Text -> FilePath
+      toHaskellFile moduleName = foldl1 (</>) (map T.unpack $ T.splitOn "." moduleName) ++ ".hs"
```

Comment:
Instead of letting `foldl1` fail, handle this potential error and provide a proper error message for the user!